lib/ex_machina/ecto.ex

```diff
@@ -90,9 +91,56 @@ defmodule ExMachina.Ecto do
   def params_with_assocs(module, factory_name, attrs \\ %{}) do
     module.build(factory_name, attrs)
     |> insert_belongs_to_assocs(module)
+    |> recursively_strip
+  end
+
+  defp recursively_strip(record = %{__meta__: %{__struct__: Ecto.Schema.Metadata}}) do
+    record
+    |> strip_has_assocs
+    |> drop_belongs_to_assocs
     |> drop_ecto_fields
     |> drop_fields_with_nil_values
   end
+  defp recursively_strip(record) do
+    record
+  end
+
+  defp strip_has_assocs(record = %{__struct__: struct}) do
+    Enum.reduce(struct.__schema__(:associations), record, fn(association_name, record) ->
+      case struct.__schema__(:association, association_name) do
+        %{__struct__: Ecto.Association.Has} ->
+          record
+          |> Map.get(association_name)
+          |> strip_assoc(record, association_name)
+        _ ->
+          record
+      end
+    end)
+  end
+
+  defp strip_assoc(original_assoc, record, association_name) do
+    case original_assoc do
+      %{__meta__: %{__struct__: Ecto.Schema.Metadata, state: :built}} ->
+        assoc = recursively_strip(original_assoc)
+        Map.put(record, association_name, assoc)
+      %{__struct__: Ecto.Association.NotLoaded} ->
+        Map.delete(record, association_name)
+      _list ->
```

Comment:
Have you thought about adding empty line before every pattern to be matched? I't might make it a little bit easier to read, especially, when there are multiple patterns (like in this case).